MobileBuy/buy/src/main/java/com/shopify/buy3/GraphClient.java

```diff
@@ -0,0 +1,137 @@
+package com.shopify.buy3;
+
+import java.util.concurrent.TimeUnit;
+
+import okhttp3.Call;
+import okhttp3.HttpUrl;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+
+public final class GraphClient implements GraphCallFactory {
+  private final HttpUrl serverUrl;
+  private final Call.Factory httpCallFactory;
+
+  private GraphClient(final String shopDomain, final String apiKey, final String applicationName, final long httpConnectionTimeoutMs,
+                      final long httpReadWriteTimeoutMs, final Interceptor[] httpInterceptors) {
+    this.serverUrl = HttpUrl.parse("https://" + shopDomain + "/api/graphql");
+    this.httpCallFactory = buildHttpCallFactory(apiKey, applicationName, httpConnectionTimeoutMs, httpReadWriteTimeoutMs, httpInterceptors);
+  }
+
+  @Override public GraphCall<APISchema.QueryRoot> queryGraph(final APISchema.QueryRootQuery query) {
+    return new RealGraphCall<>(query, serverUrl, httpCallFactory, response -> new APISchema.QueryRoot(response.getData()));
+  }
+
+  @Override public GraphCall<APISchema.Mutation> mutateGraph(final APISchema.MutationQuery query) {
+    return new RealGraphCall<>(query, serverUrl, httpCallFactory, response -> new APISchema.Mutation(response.getData()));
+  }
+
+  private OkHttpClient buildHttpCallFactory(final String apiKey, final String applicationName, final long httpConnectionTimeoutMs,
+                                            final long httpReadWriteTimeoutMs, final Interceptor[] httpInterceptors) {
+    final Interceptor requestInterceptor = chain -> {
+      final Request original = chain.request();
+      final Request.Builder builder = original.newBuilder().method(original.method(), original.body());
```

Comment:
can we get the server updated so we can pass the auth as part of the payload instead?  I know there was talk about at least doing that for customer credentials.